agir/front/components/globalContext/Toast.js

```diff
@@ -128,15 +129,38 @@ export const Toast = (props) => {
 };
 
 const ConnectedToast = (props) => {
-  const toasts = useSelector(getToasts);
+  const allToasts = useSelector(getToasts);
+  const user = useSelector(getUser);
   const dispatch = useDispatch();
   const handleClear = useCallback(
     (toastId) => {
       dispatch(clearToast(toastId));
     },
     [dispatch]
   );
-  return <Toast {...props} toasts={toasts} onClear={handleClear} />;
+
+  const [softLoginToastId, toasts] = useMemo(() => {
+    let softLoginToastId = null;
+    const toasts = allToasts.filter((toast) => {
+      if (toast.tags.includes("ANONYMOUS_TO_SOFT_LOGIN_CONNECTION")) {
+        softLoginToastId = toast.toastId;
+        return false;
+      }
+      return true;
+    });
+    return [softLoginToastId, toasts];
+  }, [allToasts]);
+
+  return (
+    <>
+      <SoftLoginModal
+        user={user}
+        shouldShow={!!softLoginToastId}
+        onClose={() => handleClear(softLoginToastId)}
+      />
```

Comment:
Cela éviterait de charger ce composant sur tous les appels ?
```suggestion
      {!!softLoginToastId && <SoftLoginModal
        user={user}
        shouldShow={!!softLoginToastId}
        onClose={() => handleClear(softLoginToastId)}
      />}
```

Reply:
Vu que c'est une modale, on ne peut pas l'ajouter uniquement quand elle est visible, sinon on perd l'animation d'entrée/sortie. Et c'est un tout petit composant en réalité.